src/library/autodj/autodjfeature.cpp

```diff
@@ -308,8 +310,22 @@ void AutoDJFeature::slotAddRandomTrack(bool) {
         retriveAttempts += 1;
     }
 
-    if (retriveAttempts == kMaxRetiveAttempts)
-        qDebug () << "Could Not Load Random Tracks";
+    if (retriveAttempts == kMaxRetiveAttempts || iTrackId == -1) {
+        qDebug () << "Could Not Load Random Tracks From Crate."
+    	        " Attempting to load from library";
+    	iTrackId = m_autoDjCratesDao.getRandomTrackIdFromLibrary();
```

Comment:
Both ways are OK for me, since the user will add a crate anyway if he uses this feature heavily.   
We will probably do the next iteration of this when @kain88-de introduce the track selector feature. 
@kain88-de: Anything to consider here in advanced?